datafusion/core/src/physical_optimizer/join_selection.rs

```diff
@@ -68,6 +75,19 @@ fn should_swap_join_order(left: &dyn ExecutionPlan, right: &dyn ExecutionPlan) -
     }
 }
 
+fn supports_collect_by_size(
+    plan: &dyn ExecutionPlan,
+    collection_size_threshold: usize,
+) -> bool {
+    if let Some(size) = plan.statistics().total_byte_size {
+        size < collection_size_threshold
+    } else if let Some(row_count) = plan.statistics().num_rows {
+        row_count < collection_size_threshold
```

Comment:
Question: are we treating both bytes and rows equally here? Seems like `collection_size_threshold` has a unit of `bytes` compared to the `row_count` which specifies number of rows. I guess we can normalize it a bit if we want to pursue this (e.g. `collection_size_threshold / SOME_MAGIC_CONSTANT`) but otherwise it might be a bit off from a desirable scenerio.

Reply:
My original thinking is making the threshold configuration represent both 1M in bytes or 1M number of rows.
Or maybe we can keep two different configuration items explicitly and document them clearly..